module-generation/frontmatter.py

```diff
@@ -0,0 +1,73 @@
+import os
+import json
+
+moduleDirs = "./scrape-data/"
+moduleDst = "./modules/"
+os.mkdir(moduleDst)
+
+for moduleDir in os.listdir(moduleDirs):
+    moduleFile = moduleDirs+moduleDir+"/module.txt"
+    getModuledata = open(moduleFile, "r")
+    readModuleData = getModuledata.read()
+    parseData = json.loads(readModuleData)
+    # Fetch module information and write on index.md file
+    try:
+        moduleName = parseData['id']
+        moduleDescription = parseData['description']
+        os.mkdir(moduleDst+moduleName)
+        indexMd = open(moduleDst+moduleName+"/index.md", "a+")
+        indexMd.write('---\n')
+        indexMd.write('posttype: "module" \n')
+        indexMd.write('title: '+moduleName+'\n')
+        indexMd.write('description: "'+moduleDescription+'"\n')
+
+        # get module cover image, write on index.md file and copy to paticular module folder
+        moduleLogo = moduleDirs+moduleDir+"/cover.png"
+        indexMd.write('cover: "./cover.png"'+'\n')
+        sourceImage = open(moduleLogo, "rb")
```

Comment:
I'd also recommend to use the `with open` here and two lines below. thus, you don't have to manually take care you close them, but python does this automagically once you leave the inner scope

Reply:
done